### PR TITLE
2465: Push Notifications in E2e Tests

### PR DESCRIPTION
**Pull request by LeandraH**
### Short description

The plan for this PR was to check that the app requests permission to send notifications, and to check if the app subscribes to the correct location and language. Unfortunately, I didn’t manage to access the console logs in the e2e environment so this PR only enables notifications in e2e and accepts the request. That makes it a bit smaller than I was hoping to do but it means that the tests will fail if the permission to send notifications is not requested. That should catch some of the notification problems.

### Proposed changes

- Enable push notifications in the e2e environment
- Allow push notifications during the build
- Completely wipe iOS devices before each test because the permissions in iOS aren't reset with a new installation (they are reset in Android when you re-install an app)

### Side effects

- The iOS tests are even slower because the device needs to be wiped between test runs

### Resolved issues

Fixes: #2465 

---

### CODE REVIEW

File: e2e-tests/native/wdio-ios.conf.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ const iosCapabilities: Capabilities.DesiredCapabilities = {
   'appium:orientation': 'PORTRAIT',
   'appium:automationName': 'XCUITest',
   'appium:language': 'EN',
+  'appium:fullReset': true,
```
Comment (LeandraH) [inline e2e-tests/native/wdio-ios.conf.ts:13]: A second installation of an app in iOS doesn't reset the permissions; without this full reset, only one in the four tests would request permission to send notifications.

Comment (steffenkleinle) [reply e2e-tests/native/wdio-ios.conf.ts:13]: :wrench: Please add a comment in the code to explain this :)

**Comment by steffenkleinle** [comment]
> The plan for this PR was to check that the app requests permission to send notifications, and to check if the app subscribes to the correct location and language. Unfortunately, I didn’t manage to access the console logs in the e2e environment so this PR only enables notifications in e2e and accepts the request. That makes it a bit smaller than I was hoping to do but it means that the tests will fail if the permission to send notifications is not requested. That should catch some of the notification problems.

Did you try to actually test/assert on push notifications? It seems like this should be possible according to some blogposts (e.g. this one for android). Or was it just too much effort?

**Comment by LeandraH** [comment]
> > The plan for this PR was to check that the app requests permission to send notifications, and to check if the app subscribes to the correct location and language. Unfortunately, I didn’t manage to access the console logs in the e2e environment so this PR only enables notifications in e2e and accepts the request. That makes it a bit smaller than I was hoping to do but it means that the tests will fail if the permission to send notifications is not requested. That should catch some of the notification problems.
> 
> Did you try to actually test/assert on push notifications? It seems like this should be possible according to some blogposts (e.g. this one for android). Or was it just too much effort?

I spent an hour or two researching it but didn't actually try. I also came to the conclusion that it is possible to test notifications directly but that it would be fragile because of secrets changing and the CMS resetting, so I decided to not add those. That is also what Steffi wrote in the issue description ("A more complex, but also more flaky way would be to really send a push notification and check if it is received. But we decided this would be not a really good solution.") but I have to confess that I don't remember that discussion.

**Comment by steffenkleinle** [comment]
> > > The plan for this PR was to check that the app requests permission to send notifications, and to check if the app subscribes to the correct location and language. Unfortunately, I didn’t manage to access the console logs in the e2e environment so this PR only enables notifications in e2e and accepts the request. That makes it a bit smaller than I was hoping to do but it means that the tests will fail if the permission to send notifications is not requested. That should catch some of the notification problems.
> > 
> > 
> > Did you try to actually test/assert on push notifications? It seems like this should be possible according to some blogposts (e.g. this one for android). Or was it just too much effort?
> 
> I spent an hour or two researching it but didn't actually try. I also came to the conclusion that it is possible to test notifications directly but that it would be fragile because of secrets changing and the CMS resetting, so I decided to not add those. That is also what Steffi wrote in the issue description ("A more complex, but also more flaky way would be to really send a push notification and check if it is received. But we decided this would be not a really good solution.") but I have to confess that I don't remember that discussion.

Thanks for explaining, sounds good to me.

**Review by steffenkleinle — APPROVED** [review]

**Review by f1sh1918 — APPROVED** [review]
finally we have e2e tests for pn. Nicely done :)

**Pull request merged**